pkg/networkmanager/firewall.jsx

```diff
@@ -210,63 +385,95 @@ class AddServicesBody extends React.Component {
+                                                                         onChange={this.onFilterChanged} />
+                                                        </td>
+                                                    </tr>
+                                                </tbody>
+                                            </table>
+                                            <ListView className="dialog-list-ct">
+                                                {
+                                                    services.map(s => (
+                                                        <ListView.Item key={s.id}
+                                                                       checkboxInput={ <input data-id={s.id}
+                                                                                              type="checkbox"
+                                                                                              checked={this.state.selected.has(s.id)}
+                                                                                              onChange={this.onToggleService} /> }
+                                                                       heading={ <React.Fragment><span>{s.name}</span>{ renderPorts(s.ports) }</React.Fragment> }>
+                                                        </ListView.Item>
+                                                    ))
+                                                }
+                                            </ListView>
+                                        </React.Fragment>
+                                    ) : (
+                                        <div className="spinner spinner-lg" />
+                                    )}
+                                </React.Fragment>
+                            }
+                            <div />
+                            <input type="radio" name="type" value="ports" onChange={this.onToggleType} disabled={this.state.avail_services == null} />
+                            {_("Custom Ports")}
+                            { !this.state.custom ||
+                                <fieldset className="ct-form-layout">
+                                    <label className="control-label" htmlFor="hint" hidden>Hint</label>
+                                    <p id="hint">
+                                        {_("Comma-separated ports, ranges and aliases are accepted")}
+                                    </p>
+
+                                    <label className="control-label" htmlFor="tcp-ports">TCP</label>
+                                    <input id="tcp-ports" type="text" onChange={this.validate}
+                                           className={"form-control " + (this.state.tcp_error ? "error" : "") }
+                                           value={this.state.custom_tcp_value}
+                                           placeholder={_("Example: 22,ssh,8080,5900,5910")} />
```

Comment:
perhaps show the last one as a range, to illustrate what a "range" is? It's otherwise not clear what the separator is. `5900-5910` sounds reasonably realistic, and by themselves they don't introduce a new concept right now.